pubsub/protocol.py

```diff
@@ -17,7 +22,44 @@ def publish(self, topic, message):
         serialized = self.serializer.encode(message=message)
         self.adapter.publish(topic, serialized)
 
-    def subscribe(self, topic):
-        for message in self.adapter.subscribe(topic):
-            serialized = self.serializer.decode(message)
-            yield serialized
+    def subscribe(self, topic, callback=None, exception_handler=lambda x, y: None, always_raise=True):
```

Comment:
Shouldn't this signature be extended to all pubsub drivers?

Reply:
Definitely